tests/utils/stl/test/Test.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+from lit.Test import Test
+
+import stl
+from stl.compiler import CXXCompiler
+
+_compiler_path_cache = dict()
+
+
+class StlTest(Test):
+    def __init__(self, suite, path_in_suite, lit_config, test_config,
+                 envlst_entry, env_num, default_cxx, file_path=None):
+        self.env_num = env_num
+        Test.__init__(self, suite, path_in_suite, test_config, file_path)
+        self._configure_cxx(lit_config, envlst_entry, default_cxx)
+
+        source_path = self.getSourcePath()
+        test_name = suite.getSourcePath(path_in_suite) + '.' + str(env_num)
+
+        self.expected_result = None
+        if test_name in test_config.expected_results:
+            self.expected_result = test_config.expected_results[test_name]
+        elif test_name in lit_config.expected_results:
+            self.expected_result = lit_config.expected_results[test_name]
+        elif source_path in test_config.expected_results:
+            self.expected_result = test_config.expected_results[source_path]
+        elif source_path in lit_config.expected_results:
+            self.expected_result = lit_config.expected_results[source_path]
+
+        if self.expected_result is not None and self.expected_result.isFailure:
+            self.xfails = ['*']
+
+    def getExecPath(self):
+        return '.'.join((Test.getExecPath(self), str(self.env_num)))
+
+    def getFullName(self):
+        return '.'.join((Test.getFullName(self), str(self.env_num)))
+
+    def _configure_cxx(self, lit_config, envlst_entry, default_cxx):
+        env_compiler = envlst_entry.getEnvVal('PM_COMPILER', 'cl')
+
+        if not os.path.isfile(env_compiler):
+            cxx = _compiler_path_cache.get(env_compiler, None)
+
+            if cxx is None:
+                search_paths = self.config.environment['PATH']
+                cxx = stl.util.which(env_compiler, search_paths)

```

Comment:
why not shutil?

Reply:
Taken from libcxx which supports python2. Looks like I can use shutil, thanks.